### PR TITLE
Crd for usage status #543

### LINKED ISSUE #543: Usage status CRD

### PR DESCRIPTION
**Pull request by MatteoGuarnaccia5**
## Description

Added table view for usage status, and CRD logic

## Testing instructions

Add a set up instructions describing how the reviewer should test the code

- [ ] Review code
- [ ] Check Actions build
- [ ] Review changes to test coverage

## Agile board tracking

closes #543

### CODE REVIEW

**Comment by MatteoGuarnaccia5** [comment]
api e2e tests do not pass in this PR as usage statuses have not been integrated. They have been integrated in #492 which will be merged into this.

File: src/admin/usageStatus/usageStatus.component.tsx
Comment (joelvdavies) [inline src/admin/usageStatus/usageStatus.component.tsx:1]: There is currenly a gap at the bottom of the table on scigateway:
image
vs
image

File: src/admin/usageStatus/deleteUsageStatusDialog.component.tsx
Comment (joelvdavies) [inline src/admin/usageStatus/deleteUsageStatusDialog.component.tsx:47]: I don't think this is very user friendly as `The specified usage status is a part of an Item. Please delete the Item first`. Perhaps
```suggestion
            setFormError(`This usage status is currently used by one or more items. Remove all uses before deleting it here.`);
```

File: src/admin/admin.component.tsx
Comment (joelvdavies) [inline src/admin/admin.component.tsx:159]: As mentioned on https://github.com/ral-facilities/inventory-management-system/pull/535 there will likely be a period where this is on dev without the units pr so it would be nice if it just displays a work in progress message.

File: src/admin/usageStatus/usageStatus.component.tsx
Comment (joelvdavies) [inline src/admin/usageStatus/usageStatus.component.tsx:1]: To fit better with the other items & units etc could this also be called `usageStatuses.component.tsx`, same for the test file.

File: src/api/units.tsx
Comment (joelvdavies) [inline src/api/units.tsx:1]: I don't think these changes are supposed to be here, they are in https://github.com/ral-facilities/inventory-management-system/pull/542.

File: src/mocks/Units.json
Comment (joelvdavies) [inline src/mocks/Units.json:1]: I think this is only supposed to be in https://github.com/ral-facilities/inventory-management-system/pull/542.

File: src/mocks/handlers.ts
Comment (joelvdavies) [inline src/mocks/handlers.ts:801]: From https://github.com/ral-facilities/inventory-management-system/pull/542.

**Review by joelvdavies — CHANGES_REQUESTED** [review]
Initial review, I think some changes got pulled over from https://github.com/ral-facilities/inventory-management-system/pull/542.

**Comment by MatteoGuarnaccia5** [comment]
> Initial review, I think some changes got pulled over from #542.

I initially had branched off of #542 but then changed the base - that is probably why things for units are in this PR. I can just remove them it has no impact on this PR

**Comment by joelvdavies** [comment]
> > Initial review, I think some changes got pulled over from #542.
> 
> I initially had branched off of #542 but then changed the base - that is probably why things for units are in this PR. I can just remove them it has no impact on this PR

That would be good, thanks. We will be running this on dev as part of the migration process but without any of the units backend changes as we will migrate usage status' first and then the units separately afterwards.

File: src/admin/admin.component.tsx
Comment (MatteoGuarnaccia5) [reply src/admin/admin.component.tsx:159]: I've left the `units.component.tsx` file with just a work in progress message like in #535

File: src/admin/usageStatuses/usageStatusDialog.component.tsx
Comment (joelvdavies) [inline src/admin/usageStatuses/usageStatusDialog.component.tsx:78]: Could this width be reduced, it seems a bit too wide for what it is
image
perhaps md or sm look better?

File: src/admin/usageStatuses/__snapshots__/usageStatus.component.test.tsx.snap
Comment (joshuadkitenge) [inline src/admin/usageStatuses/__snapshots__/usageStatus.component.test.tsx.snap:1]: can you delete this snapshot

File: src/admin/usageStatuses/deleteUsageStatusDialog.component.tsx
Comment (joshuadkitenge) [inline src/admin/usageStatuses/deleteUsageStatusDialog.component.tsx:63]:
```suggestion
        Delete Usage Status
```

File: src/admin/usageStatuses/deleteUsageStatusDialog.component.tsx
Comment (joshuadkitenge) [inline src/admin/usageStatuses/deleteUsageStatusDialog.component.tsx:67]:
```suggestion
        <strong data-testid="delete-usage-status-value">{usageStatus?.value}</strong>?
```

File: src/admin/usageStatuses/deleteUsageStatusDialog.component.tsx
Comment (joshuadkitenge) [inline src/admin/usageStatuses/deleteUsageStatusDialog.component.tsx:72]:
```suggestion
          onClick={handleDeleteUsageStatus}
```

File: src/admin/usageStatuses/usageStatusDialog.component.test.tsx
Comment (joshuadkitenge) [inline src/admin/usageStatuses/usageStatusDialog.component.test.tsx:10]:
```suggestion
describe('Usage status dialog', () => {
```

File: src/admin/usageStatuses/usageStatusDialog.component.test.tsx
Comment (joshuadkitenge) [inline src/admin/usageStatuses/usageStatusDialog.component.test.tsx:36]:
```suggestion
  it('displays errors correctly', async () => {
```

File: src/admin/usageStatuses/usageStatusDialog.component.tsx
Comment (joshuadkitenge) [inline src/admin/usageStatuses/usageStatusDialog.component.tsx:83]:
```suggestion
              sx={{ marginLeft: '4px', my: '8px' }}
```

File: src/app.types.tsx
Comment (joshuadkitenge) [inline src/app.types.tsx:357]:
```suggestion
export interface AddUsageStatus {
  value: string;
}

export interface UsageStatus extends AddUsageStatus {
  id: string;
  code: string;
  created_time: string;
  modified_time: string;
}
```

File: src/items/itemDialog.component.tsx
Comment (joshuadkitenge) [inline src/items/itemDialog.component.tsx:502]:
```suggestion
    details,
```

**Review by joshuadkitenge — CHANGES_REQUESTED** [review]
Works well with some minor comments 

- there are some unit test warning in the usageStatusDialog

File: cypress/e2e/with_mock_data/items.cy.ts
Comment (joelvdavies) [inline cypress/e2e/with_mock_data/items.cy.ts:218]:
```suggestion
  it('adds an item with only mandatory fields (allowed list of values)', () => {
```

File: src/mocks/handlers.ts
Comment (joelvdavies) [inline src/mocks/handlers.ts:854]:
```suggestion
            detail: 'The specified usage status is a part of an Item',
```

File: src/mocks/handlers.ts
Comment (joelvdavies) [inline src/mocks/handlers.ts:818]:
```suggestion
            detail: 'A Usage Status with the same value already exists',
```

File: .github/workflows/ci-build.yml
Comment (joelvdavies) [inline .github/workflows/ci-build.yml:122]:
```suggestion
          harbor.stfc.ac.uk/inventory-management-system/ims-api:develop
```
Before merging.

**Review by joelvdavies — APPROVED** [review]
LGTM - Don't merge this PR, we will once its successfully running on the dev machine.

**Review by joshuadkitenge — APPROVED** [review]
LGTM

File: src/systems/systemItemsTable.component.tsx
Comment (joshuadkitenge) [inline src/systems/systemItemsTable.component.tsx:330]:
```suggestion
       accessorFn:
          type === 'usageStatus' ? undefined : (row) => row.item.usage_status,
```
This disables the column filtering for usage status in the systemsItemDialog. This is needed because the filtered data and the data in the table is going to different. (pre-existing usage status data and inputted usages status data )

**Review by joshuadkitenge — APPROVED** [review]
LGTM

**Review by joelvdavies — DISMISSED** [review]

**Review by joelvdavies — APPROVED** [review]

**Review by joshuadkitenge — APPROVED** [review]

**Pull request merged**
